app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/MainActivityTest.kt

```diff
@@ -27,24 +31,71 @@ class MainActivityTest {
     @get:Rule(order = 1)
     val testRule = ActivityScenarioRule(MainActivity::class.java)
 
+    @get:Rule(order = 2)
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @Test
+    fun navigateHomePageAddsTheCorrectFragment() {
+        onView(withId(R.id.activityMainBottomNavigationView)).perform(navigateTo(R.id.homeNavItem))
+        testRule.scenario.onActivity { activity ->
+            val fragment: Fragment? =
+                activity.supportFragmentManager.findFragmentById(R.id.mainActivityFragmentContainer)
+            assertEquals(true, fragment is HomeFragment)
+        }
+    }
+
     @Test
-    fun firesAnIntentWhenUserPressesCourseButton() {
+    fun navigateToReviewAddsTheCorrectFragment() {
+        onView(withId(R.id.activityMainBottomNavigationView)).perform(navigateTo(R.id.reviewNavItem))
+        testRule.scenario.onActivity { activity ->
+            val fragment: Fragment? =
+                activity.supportFragmentManager.findFragmentById(R.id.mainActivityFragmentContainer)
+            assertEquals(true, fragment is ReviewFragment)
+        }
+    }
+
+    @Test
+    fun navigateToEventPageAddsTheCorrectFragment() {
+        onView(withId(R.id.activityMainBottomNavigationView)).perform(navigateTo(R.id.eventNavItem))
+        testRule.scenario.onActivity { activity ->
+            activity.findViewById<BottomNavigationView>(R.id.activityMainBottomNavigationView).selectedItemId =
```

Comment:
This line is doing the same thing as navigateTo ? Or I misunderstood something ?

Reply:
(line 61)

Reply:
Forgot to remove it. Thanks for the notice ;)